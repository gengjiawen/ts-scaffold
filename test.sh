set -e

# for macOS
# yes "" | ./node_modules/.bin/vue init . test

echo -ne '\n' | ./node_modules/.bin/vue init . test

cd test
npm install
npm run build
