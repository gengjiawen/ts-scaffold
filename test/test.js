import path from 'path'
import test from 'ava'
import sao from 'sao'
import execa from 'execa'

const generator = path.join(__dirname, '..')

test('defaults', async t => {
  const stream = await sao.mock({ generator })

  t.snapshot(stream.fileList, 'Generated files')
})

const log = i => {
  console.log(i.stdout)
  console.log(i.stderr)
}

test('e2e', async t => {
  const outDir = path.join(__dirname, '..', 'sample_out')
  await execa(`npx sao ${generator} ${outDir} -y`, {
    cwd: path.join(__dirname, '..'),
  })
    .then(log)
  await execa(`yarn`, {
    cwd: outDir,
  })
    .then(log)
  await execa(`yarn test`, {
    cwd: outDir,
  })
    .then(log)
})
