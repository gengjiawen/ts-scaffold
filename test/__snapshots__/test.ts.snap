// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all files generated 1`] = `
Array [
  ".github/workflows/CI.yml",
  ".gitignore",
  ".gitlab-ci.yml",
  ".gitpod.yml",
  ".npmignore",
  ".prettierignore",
  ".prettierrc",
  ".vscode/extensions.json",
  "LICENSE",
  "README.md",
  "libs/index.test.ts",
  "libs/index.ts",
  "package.json",
  "tsconfig.json",
]
`;
